added task1 code and Cmake file.

VadymSalymonenko opened the pull request:
Added two parts of task 1. They are implemented in the main.c file.

VadymT95 committed:
added task1 code and Cmake file.

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+        return 1;
+    }
+
+    int elf_calories[TOP_LIST_SIZE] = {0};
+    int elf_calories_numbers[TOP_LIST_SIZE] = {0};
```
change this to the struct, aka:
// Struct to store the elf's ID and total calories.
typedef struct {
    unsigned int id;
    unsigned int totalCalories;
} Elf;

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+    int elf_calories[TOP_LIST_SIZE] = {0};
+    int elf_calories_numbers[TOP_LIST_SIZE] = {0};
+
+    int curr_total_food = 0;
+    int curr_elf_number = 0;
```
unsigned int currTotalCalories = 0;
 unsigned int currElfId = 1;  // Start from 1 for clarity.

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+            curr_elf_number++;
+            curr_total_food = 0;
+        } else {
+            int input_num = atoi(line);
+            curr_total_food += input_num;
```
currTotalCalories += atoi(line);

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+    printf("total sum: %d\n", sum);
+}
+
+int find_insertion_index(int arr[], int size, int key) {
```
```suggestion
unsigned int find_insertion_index(const Elf topElves[], unsigned int newCalories) {
```

Simplify the insertion logic by iterating through the list rather than using binary search, since our list size is small.

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+        printf("place %d: calories -- %d, id: %d\n", size-i+1, elf_calories[i-1], elf_calories_numbers[i-1]);
+    }
+}
+void print_sum_of_elements(int arr[], int size) {
```
```suggestion
void print_total_calories(const Elf topElves[]) {
```

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+#define MAX_INT_NUMBER_LENGTH 12
+#define TOP_LIST_SIZE 3
+
+void print_top_list(int* elf_calories, int* elf_calories_numbers, int size){
+    for(int i = size; i > 0; i--){
+        printf("place %d: calories -- %d, id: %d\n", size-i+1, elf_calories[i-1], elf_calories_numbers[i-1]);
+    }
+}
```
```suggestion
void print_top_elves(const Elf topElves[]) {
    for (int i = TOP_LIST_SIZE - 1; i >= 0; i--) {
        printf("Place %d: Elf ID -- %d, Calories -- %d\n", TOP_LIST_SIZE - i, topElves[i].id, topElves[i].totalCalories);
    }
}
```

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+    return start;
+}
+
+void update_elf_calories(int* elf_calories, int* elf_calories_numbers, int size, int new_calories, int curr_elf_number) {
```
```suggestion
void update_top_elves(Elf topElves[], unsigned int newCalories, unsigned int currElfId) {
```

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+
+    if (new_calories <= elf_calories[0]) {
+        return;
+    }
```
remove this

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+    for (int i = 1; i < insertion_index; i++) {
+        elf_calories[i - 1] = elf_calories[i];
+        elf_calories_numbers[i - 1] = elf_calories_numbers[i];
+    }
```
```suggestion
   unsigned int idx = find_insertion_index(topElves, newCalories);
    if (idx < TOP_LIST_SIZE) {
        for (int i = 0; i < idx; i++) {
            topElves[i] = topElves[i + 1];
        }
```

BBuinich commented on 1/src/main.c:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    elf_calories[insertion_index - 1] = new_calories;
+    elf_calories_numbers[insertion_index - 1] = curr_elf_number;
```
use idx instead of _insertion_index - 1_

BBuinich reviewed (changes requested):
looks good to me, please check my comments

VadymT95 committed:
Small optimization.

1)creating a struct to connection an elf's number with his calories;
2)minor changing variables and functions.

BBuinich reviewed (approved):
Merge it
